pkg/registry/core/service/storage/rest_test.go

```diff
@@ -2437,3 +2449,59 @@ func TestServiceRegistryCreateWithoutExistingNetwork(t *testing.T) {
 		t.Fatalf("expected error; got none")
 	}
 }
+
+func TestServiceRegistryCreateWithNetworkOfArktosIPAM(t *testing.T) {
+	origFeatureGate := newFeatureGateWithPerNetworkServiceIPAlloc()
+	defer func() {
+		utilfeature.DefaultFeatureGate = origFeatureGate
+	}()
+
+	storage, registry, server := NewTestREST(t, nil)
+	storage.networks = &dummyNetworkGetter{ipamType: "Arktos", cidr: "3.4.5.0/24"}
+	_, ipRange, _ := net.ParseCIDR("3.4.5.0/24")
+	defer server.Terminate(t)
+
+	svc := &api.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Tenant:    "system",
```

Comment:
As we are working on multi-tenancy, it is better test on a non-system tenant.

Reply:
got your point. However, this case is purposed as the most basic scenario for feature of per-network service ip allocation; in case it fails, we know for sure it is broken at this feature, not around multi-tenancy.